pkg/controller/discovery/container/ds.go

```diff
@@ -209,7 +210,11 @@ func (r *DataSource) buildClient(cluster *migapi.MigCluster) error {
 		Log.Trace(err)
 		return err
 	}
-	r.Client, err = cluster.GetClient(r.Container.Client)
+	r.Client, err = client.New(
```

Comment:
Each `DataSource` needs it's own _client_.  The MigCluster will return the same client when IsHostCluster=true.  I have seen this cause problems when a client is used concurrently.